rules/alerting_test.go

```diff
@@ -36,16 +36,19 @@ import (
 	"github.com/prometheus/prometheus/util/testutil"
 )
 
-var testEngine = promql.NewEngine(promql.EngineOpts{
```

Comment:
This kind of reverts the changes in https://github.com/prometheus/prometheus/pull/12050/files#diff-da69847bdad5d79a80d083f670ffdc95b2202bb3edbc580fb777bc4bf5b0081cR36, I don't know if having a shared engine was the goal or just a consequence/improvement.

Reply:
I doubt that sharing state between tests is a good thing, since as a rule of thumb tests should be independent of each other. Plus, you can't clean up `testEngine` after you're done testing. I'll ask @bwplotka if he sees a problem with making `testEngine` into a function.

Reply:
@machine424 @MichaHoffmann [said it's OK](https://cloud-native.slack.com/archives/C01AUBA4PFE/p1725288113941779) :)